torch/csrc/autograd/python_function.cpp

```diff
@@ -609,8 +609,11 @@ PyObject *THPFunction_apply(PyObject *cls, PyObject *inputs)
   HANDLE_TH_ERRORS
   auto info_pair = unpack_input<false>(inputs);
   auto iv_inputs = std::vector<c10::IValue>();
-  for (auto i : info_pair.first.input_vars) {
-     iv_inputs.push_back(c10::IValue(i));
+  bool record = false;
+  if (C10_UNLIKELY(at::shouldRunRecordFunction(&record))) {
+    for (auto i : info_pair.first.input_vars) {
+       iv_inputs.push_back(c10::IValue(i));
+    }
   }
   RECORD_FUNCTION(
```

Comment:
@moderato See https://github.com/pytorch/pytorch/blob/38c185189c4e7b11f9bd9b60afed75b367ee00e2/aten/src/ATen/record_function.h#L464-L472

The same check is repeated. So It might make sense to use the C++ API directly rather than using the macro.

Reply:
Done.